### PR TITLE
[SP] Add example section in SP spec

### PR DESCRIPTION
**Pull request by DonnaWuDongxia**
BUG=XWALK-6402

### CODE REVIEW

**Comment by DonnaWuDongxia** [comment]
@huningxin @halton Please help to review.

**Comment by huningxin** [comment]
Raw git link for easier review: https://rawgit.com/DonnaWuDongxia/realsense-extensions-crosswalk/specExample/spec/scene-perception.html#examples

File: spec/scene-perception.html [hunk 1]
```diff
@@ -1194,6 +1194,146 @@
         </dl>
       </section>
     </section>
+    <section class='informative'>
+      <h2>
+        Examples
+      </h2>
+      <h3>
```
Comment (huningxin) [inline spec/scene-perception.html]: Please add `<section>` for each sample like DEP spec.

File: spec/scene-perception.html [hunk 2]
```diff
@@ -1194,6 +1194,146 @@
...
+        Scene Perception(SP) module configuration.
```
Comment (huningxin) [inline spec/scene-perception.html]: How about "Configure and Control Scene Perception module"?

File: spec/scene-perception.html [hunk 3]
```diff
@@ -1194,6 +1194,146 @@
...
+
+        destroyButton.onclick = function(e) {
+          sp.destroy().then(function() {
+            console.log('stop succeeds');
+            resetButtonState(true);
+          }, function(e) {console.log(e);});
+        };
+      </pre>
+      <h3>
+        Get data from SP module by binding event handlers.
```
Comment (huningxin) [inline spec/scene-perception.html]: How about "Handle events of SP module"?

File: spec/scene-perception.html [hunk 4]
```diff
@@ -1194,6 +1194,146 @@
...
+          qualityElement.innerHTML = 'Quality: ' + quality.toFixed(2);
+        };
+
+        // Please refer the SampleProcessedEvent inferface for event data.
+        sp.onsampleprocessed = function(e) {
+          accuracyElement.innerHTML = 'Accuracy: ' + e.data.accuracy;
+          qualityElement.innerHTML = 'Quality: ' + e.data.quality.toFixed(2);
+          sp.getSample().then(function(sample) {
+            // Show the color and depth image, please refer getSample interface for more details.
+            showColorImage(sample.color);
```
Comment (huningxin) [inline spec/scene-perception.html]: As we won't provide the code of `showColorImage` and `showDepthImage`. You can just put some comments to give audience the idea about the data to be used. Like https://rawgit.com/jondong/realsense-extensions-crosswalk/master/spec/depth-enabled-photography.html#access-photo-information, the description of color image and depth image.

File: spec/scene-perception.html [hunk 5]
```diff
@@ -1194,6 +1194,146 @@
...
+            showDepthImage(sample.depth);
+          }, function(e) {console.log(e);});
+          sp.queryVolumePreview(e.data.cameraPose).then(function(volumePreview) {
+            // Show the volume preview image.
+            // Please refer VolumePreviewData interface for detail data format.
+            showVolumePreviewImage(volumePreview);
```
Comment (huningxin) [inline spec/scene-perception.html]: ditto.

File: spec/scene-perception.html [hunk 6]
```diff
@@ -1194,6 +1194,146 @@
...
+          });
+        };
+
+        sp.onmeshupdated = function(e) {
+          thisObj = this;
+          sp.getMeshData().then(function(meshes) {
+            // In the meshes, there are information about vertices, faces, color.
+            // Please refer MeshData inferface for more details.
+            var func = reconstructScene.bind(thisObj, meshes);
```
Comment (huningxin) [inline spec/scene-perception.html]: The `reconstructScene` function without code might confuse audience. I think comments are good enough.

**Comment by huningxin** [comment]
Done my review. Please take a look at the comments. Thanks!

**Comment by DonnaWuDongxia** [comment]
@huningxin updated, thanks!

File: spec/scene-perception.html [hunk 7]
```diff
@@ -1194,6 +1194,168 @@
...
+        <pre class='example highlight'>
+          var qualityElement = document.getElementById('quality');
+          var accuracyElement = document.getElementById('accuracy');
+
+          sp.onchecking = function(e) {
+            var quality = e.data.quality;
+            qualityElement.innerHTML = 'Quality: ' + quality.toFixed(2);
+          };
+
+          // Please refer the SampleProcessedEvent inferface for event data.
```
Comment (huningxin) [inline spec/scene-perception.html]: refer -> refer to

File: spec/scene-perception.html [hunk 8]
```diff
@@ -1194,6 +1194,168 @@
...
+          sp.onsampleprocessed = function(e) {
+            accuracyElement.innerHTML = 'Accuracy: ' + e.data.accuracy;
+            qualityElement.innerHTML = 'Quality: ' + e.data.quality.toFixed(2);
+
+            sp.getSample().then(function(sample) {
+              // The sample object contains color image and depth image.
+              // The image data structure, which could possibly looks like this:
```
Comment (huningxin) [inline spec/scene-perception.html]: looks -> look

File: spec/scene-perception.html [hunk 9]
```diff
@@ -1194,6 +1194,168 @@
...
+              //                  width: imageWidth,
+              //                  height: imageHeight,
+              //                  data: imageDataArrayBuffer
+              //                };
+              // sample.depth = {
+              //                  width: imageWidth,
+              //                  height: imageHeight,
+              //                  data: imageDataArrayBuffer
+              //                };
+              // The imageDataArrayBuffer is an ArrayBuffer with type of Uint32Array, and the byte
```
Comment (huningxin) [inline spec/scene-perception.html]: For color image, the data should be a Uint8Array. For depth image, the data should be a Uint16Array.

File: spec/scene-perception.html [hunk 10]
```diff
@@ -1194,6 +1194,168 @@
...
+            }, function(e) {console.log(e);});
+
+            sp.queryVolumePreview(e.data.cameraPose).then(function(volumePreview) {
+              // The volume preview image data structure is similar with color and depth image in sample.
+              // volumePreview = {
+              //                   width: imageWidth,
+              //                   height: imageHeight,
+              //                   data: imageDataArrayBuffer
+              //                 };
+              // The imageDataArrayBuffer is an ArrayBuffer with type of Uint32Array, and the byte
```
Comment (huningxin) [inline spec/scene-perception.html]: The data should be Uint8Array.

File: spec/scene-perception.html [hunk 11]
```diff
@@ -1194,6 +1194,168 @@
...
+              // layout follows the Canvas Pixel ArrayBuffer defined in [CANVAS-2D].
+              // Please refer VolumePreviewData interface for detail data format.
+            });
+          };
+
+          sp.onmeshupdated = function(e) {
+            thisObj = this;
+            sp.getMeshData().then(function(meshes) {
+              // In the meshes, there are information about vertices, faces, color.
```
Comment (huningxin) [inline spec/scene-perception.html]: faces, color -> faces and color.

File: spec/scene-perception.html [hunk 12]
```diff
@@ -1194,6 +1194,168 @@
...
+              // Meshes can be used to reconstruct the scaned scene by WebGL.
```
Comment (huningxin) [inline spec/scene-perception.html]: scaned -> scanned

File: spec/scene-perception.html [hunk 13]
```diff
@@ -1194,6 +1194,168 @@
...
+              // Please refer MeshData inferface for more details.
```
Comment (huningxin) [inline spec/scene-perception.html]: refer -> refer to

**Comment by huningxin** [comment]
Some nits, please take a look. Thanks!

**Comment by DonnaWuDongxia** [comment]
@huningxin updated, please have another look. Thanks!

**Comment by huningxin** [comment]
Great!

LGTM. Merging.

**Pull request merged**